concert/experiments/imaging.py

```diff
@@ -857,3 +861,216 @@ async def _take_radios(self):
                 yield frame
         finally:
             await self._finish_radios()
+
+
+class GratingInterferometryMixin(Parameterizable):
+    """
+    Mixin for grating interferometry specific experiments.
+    """
+
+    grating_period = Quantity(q.um)
+    """Period of the grating to scan."""
+
+    num_periods = Parameter()
+    """Number of gratings period so scan."""
+
+    num_steps_per_period = Parameter()
+    """Number of stepping positions per grating period."""
+
+    stepping_start_position = Quantity(q.um)
+    """Position of the first grating step of *stepping_motor*"""
+
+    propagation_distance = Quantity(q.mm)
+    """
+    Distance between the sample and the analyzer grating.
+
+    This is only used by the processing addon to calculate the phase shift.
+    """
+    def __init__(self, stepping_motor,
+                 grating_period, stepping_start_position, num_periods=1, num_steps_per_period=16,
+                 propagation_distance=None):
+        """
+        :param stepping_motor: Motor for the stepping of one of the gratings
+        :type stepping_motor: LinearMotor
+        :param grating_period: Period of the stepped grating
+        :type grating_period: q.um
+        :param stepping_start_position: First stepping position
+        :type stepping_start_position: q.um
+        :param num_periods: Number of periods that are sampled by the stepping
+        :type num_periods: int
+        :param num_steps_per_period: Number of stepping positions per period
+        :type num_steps_per_period: int
+        :param propagation_distance: Distance between the sample and the analyzer grating. Only used
+            by the processing addon to determine the phase shift in angles.
+        :type propagation_distance: q.mm
```

Comment:
Could you convert these to the ones which are used throughout the rest of the concert? I mean
```
.. py:attribute:: foobar
    descr

```

Reply:
Hmm, but that adds a nicely indented "parameters" section to the docs, so leave it.